docs/mypy.md

```diff
@@ -0,0 +1,30 @@
+# mypy static checker
+
+Zulip now uses mypy. mypy is a compile-time type checker for python.
+For information about what mypy is, visit [http://mypy-lang.org/](http://mypy-lang.org/).
+
+mypy's type checking has been added to Zulip's Travis CI checks.
```

Comment:
I think it's worth documenting that we're using the Python 2 style comment syntax for mypy and linking to a page describing how that works, e.g. http://mypy.readthedocs.org/en/latest/python2.html and https://www.python.org/dev/peps/pep-0484/